### ISSUE TITLE
[a11y]: NumberInput - JAWS Announces the Input label name even after it is hidden

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16531]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

From the Number input - playground, set the hidelabel Boolean to "true" (i.e `hideLabel` **prop** is `true`) and observe the JAWS announcement.
JAWS will announce the name of the input label which is visually hidden.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-numberinput--playground&args=hideLabel:!false

### Recommend Fix
Ensure that visually hidden label is hidden/not announced from/by screen reader as well

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by wkeese** [comment]
Isn't that how it's supposed to work, for labels on any input?

**Comment by 2nikhiltom** [comment]
Visually hidden `label` are expected to be announced by Screen readers - [reference](https://github.com/carbon-design-system/carbon/pull/18080)
